sym_api_client_python/datafeed_event_service.py

```diff
@@ -5,358 +5,63 @@
-
-            previous_exc: provide the exception that causes the original timeout to have that
-                          attached to the stacktrace of the runtime error, showing more clearly
-                          why the timeout occurred. """
-
-        if (self.upper_threshold is not None) and (self.current_timeout_sec > self.upper_threshold):
-            e = RuntimeError(
-                    'Upper timeout threshold exceeded: {:.4g}s'.format(self.upper_threshold)
-                )
-            if previous_exc is not None:
-                raise e from previous_exc
-            else:
-                raise e
-        else:
-            original = self.current_timeout_sec
-            new_timeout = original * self.timeout_multiplier
-            self.current_timeout_sec = new_timeout
-            log.debug('DataFeedEventService/get_and_increase_timeout --> '
-            'Using current timeout of {:.4g}s, but increasing to {:.4g}s for next time'
-            .format(original, new_timeout))
-            return max(self.lower_threshold, original)
-
-    def decrease_timeout(self):
-        original = self.current_timeout_sec
-        new_timeout = self.baseline_timeout_sec
-        if original != new_timeout:
-            log.debug('DataFeedEventService/decrease_timeout() --> '
-                'Decreasing timeout from {:.4g}s to {:.4g}s'.format(original, new_timeout))
-            self.current_timeout_sec = new_timeout
-        return self.current_timeout_sec
-
-
-    def msg_sent_handler(self, payload):
-        """This handler is used for both room messages and IMs. Which listener is invoked
-        depends on the streamType"""
-        log.debug('msg_sent_handler function started')
-        stream_type = payload['payload']['messageSent']['message']['stream']['streamType']
-        message_sent_data = payload['payload']['messageSent']['message']
-        if str(stream_type) == 'ROOM':
-            for listener in self.room_listeners:
-                listener.on_room_msg(message_sent_data)
-        elif str(stream_type) == 'POST':
-            for listener in self.wall_post_listeners:
-                listener.on_wall_post_msg(message_sent_data)
-        else:
-            for listener in self.im_listeners:
-                listener.on_im_message(message_sent_data)
+        self.datafeed_event_service.deactivate_datafeed()
 
-    def instant_msg_handler(self, payload):
-        log.debug('instant_msg_handler function started')
-        instant_message_data = payload['payload']['instantMessageCreated']
-        for listener in self.im_listeners:
-            listener.on_im_created(instant_message_data)
-
-
-    def room_created_handler(self, payload):
-        log.debug('room_created_handler function started')
-        room_created_data = payload['payload']['roomCreated']
-        for listener in self.room_listeners:
-            listener.on_room_created(room_created_data)
+    def add_room_listener(self, im_listener):
```

Comment:
`add_room_listener` should take `room_listener` as argument instead of `im_listener`

I can only see `add_room_listener` and `add_im_listener`, how about the other listeners(connections, elements,...)?

Reply:
I am going to add them.

Reply:
fixed